### PR TITLE
New Entry: R Functions

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

From Target Keyword List

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/r/concepts/functions/functions.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+```
+
+The output of the above will be:
+
+```shell
+[1] "Even."
+[2] "Odd."
+```
+
+## The `return()` Function
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:76]:
```suggestion
### The `return()` Function
```

File: content/r/concepts/functions/functions.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+print(y)
+```
+
+The output of the above will be:
+
+```shell
+[1] 2
+```
+
+## Default Parameter Values
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:100]:
```suggestion
### Default Parameter Values
```

File: content/r/concepts/functions/functions.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Functions are blocks of code that can be repeatedly called by other code. A function is not called when it is defined, but only when another part of the code executes the function.
+
+## Syntax
+
+The syntax for a function in R is slightly different than in other programming languages.
+
+```r
+my_function <- function(parameters) {
+  # Code to execute.
+}
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:25]:
```suggestion
}

my_function(parameters)
```

File: content/r/concepts/functions/functions.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+- `my_function`: The name of the function. It is a [variable](https://www.codecademy.com/resources/docs/r/variables) and obeys the naming convention for variables.
+- `parameters`: Zero or more comma-separated variables that will be passed on to the code block when the function is called.
+- `code to execute`: The code that executes when the function is called.
+
+A function is called by using its name, followed by parenthesis enclosing any parameters the function expects:
+
+```r
+my_function(parameters)
+```
+
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:37]:
```suggestion

```

File: content/r/concepts/functions/functions.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+Arguments can be passed to a function by position or by name. By position passes arguments in the order they were defined in the function. By name specifies the name of the parameter given by the function.
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:38]:
```suggestion
### Arguments

Arguments can be passed to a function by position or by name. By position passes arguments in the order they were defined in the function. By name specifies the name of the parameter given by the function.
```

File: content/r/concepts/functions/functions.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+```r
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:40]:
```suggestion
```pseudo
```

File: content/r/concepts/functions/functions.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+my_function <- function(parameters) {
+  # Code to execute.
+}
+```
+
+- `my_function`: The name of the function. It is a [variable](https://www.codecademy.com/resources/docs/r/variables) and obeys the naming convention for variables.
+- `parameters`: Zero or more comma-separated variables that will be passed on to the code block when the function is called.
+- `code to execute`: The code that executes when the function is called.
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:30]:
```suggestion
- `# Code to execute`: The code that executes when the function is called.
```

File: content/r/concepts/functions/functions.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+A function is called by using its name, followed by parenthesis enclosing any parameters the function expects:
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:32]:
```suggestion
A function is called with its name, followed by any parameters that are expected in the parentheses.
```

File: content/r/concepts/functions/functions.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+# Passing arguments by position:
+my_function(value1, value2, value3)
+
+# Passing arguments by name:
+my_function(arg1 = value1, arg2 = value2, arg3 = value3)
+```
+
+## Example
+
+The following takes a numeric argument and prints a string saying if it is even or odd:
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:54]:
```suggestion
The following example defines an `odd_or_even()` function that takes a `num` argument. When called, it prints a message saying if `num` is even or odd:
```

File: content/r/concepts/functions/functions.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+## The `return()` Function
+
+The `return()` function stops execution of the function and returns a result value. Unlike other languages, `return()` in R is a function and the result being returned must be enclosed in parenthesis.
+
+```r
+my_function <- function() {
+  x <- 1
+  x <- x + 1
+  return (x)
+  # The following lines are never executed.
```
Comment (Dusch4593) [inline content/r/concepts/functions/functions.md:85]:
```suggestion
  return (x)
  
  # The following lines are never executed.
```

**Comment by SSwiniarski** [comment]
@Dusch4593, I implemented your suggestions.

**Review by Dusch4593 — APPROVED** [review]
@SSwiniarski LGTM for the next review! 👍🏻

**Review by KTom101 — CHANGES_REQUESTED** [review]
@SSwiniarski Just a couple changes from me.

**Comment by SSwiniarski** [comment]
@KTom101, I implemented your suggestions.

**Review by KTom101 — APPROVED** [review]
@SSwiniarski 👍

**Comment by Dusch4593** [comment]
This seems ready to merge but I made some edits in b2a33093e2ec6fc15f9e5143aa7e79651ee2dc9c that include:

- Removing link to General: Functions entry (this isn't a practice I've seen with other language-specific "Function" entries).
- Changing to \```pseudo snippet in the top-level "Syntax" section.
- Adding list item about the `<-` operator used in the "Syntax" snippet.
- Changing "parenthesis" to "parentheses" (under "The `return()` Function").

cc: @SSwiniarski @KTom101

**Pull request merged**